src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -56,15 +62,35 @@ class BlockImporter(
 
   private def running(state: ImporterState): Receive = handleTopMessages(state, running) orElse {
     case ReceiveTimeout => self ! PickBlocks
+
     case PrintStatus => log.info("Block: {}, is on top?: {}", blockchain.getBestBlockNumber(), state.isOnTop)
+
     case BlockFetcher.PickedBlocks(blocks) =>
       SignedTransaction.retrieveSendersInBackGround(blocks.toList.map(_.body))
       importBlocks(blocks)(state)
+
     case MinedBlock(block) =>
       if (!state.importing) {
         importMinedBlock(block, state)
       }
+
+    case nc @ NewCheckpoint(parentHash, signatures) =>
+      if (state.importing) {
+        //TODO: is this ok? What delay?
+        context.system.scheduler.scheduleOnce(100.millis, self, nc)
```

Comment:
IMHO we can treat it like other block types and don't try to re-import it.

Reply:
What if all the federation nodes happen to be importing other blocks when a new checkpoint appears? Wouldn't it be lost forever?

Reply:
assuming that we know OBFT nodes and federation nodes, they shouldn't start producing checkpoints before being on top of the chain, so I'm ok with your restart logic. The only thing worth to consider is higher restart time, maybe something ~1 s, to give to node more time to re-sync